packages/big-design/src/utils/useDidUpdate.ts

```diff
@@ -0,0 +1,13 @@
+import { useEffect, useRef, DependencyList, EffectCallback } from 'react';
+
+export const useDidUpdate = (effect: EffectCallback, deps?: DependencyList) => {
+  const isInitialRender = useRef(true);
```

Comment:
Why is this achieved with `useRef` instead of `const [isInitialRender, setIsInitialRender] = useState(true)`? It seems to me that would achieve the same thing

Reply:
It achieves the same thing, however, calling `setIsInitialRender` triggers a re-render. So initially it would run `render` twice with no changes. Using `ref` we can mutate it directly and avoid a re-render.